src/testdir/test_functions.vim

```diff
@@ -951,8 +951,13 @@ func Test_mode()
     call assert_equal('nt', mode(1))
     call feedkeys(":echo \<C-R>=Save_mode()\<C-U>\<CR>", 'xt')
     call assert_equal("c-c", g:current_modes)
+
     call feedkeys("aexit\<CR>", 'xt')
   endif
+  call feedkeys(":\<insert>\<C-r>=Save_mode()\<CR>",'xt')
```

Comment:
Putting these just after exiting terminal won't work, because exiting terminal needs a delay. Like I previous said, I'd suggest putting these lines just below line 918, grouped with other cmdline mode test cases.